File: shoe_wash_app/backend/src/domain/item_class.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class ItemClass:
    """A category of item that can be washed, with pricing, true cost,
    and how long one wash actually takes."""

    id: Optional[int]
    name: str
    base_price: float
    unit_cost: float = 0.0       # water/power/soap/upkeep allocated per item
    wash_minutes: int = 30       # how long one wash occupies the machine

    def __post_init__(self):
        if not self.name.strip():
            raise ValueError("Item class name is required")
        if self.base_price <= 0:
            raise ValueError("Base price must be positive")
        if self.unit_cost < 0:
            raise ValueError("Unit cost cannot be negative")
        if self.wash_minutes <= 0:
            raise ValueError("Wash minutes must be positive")

File: shoe_wash_app/mobile_app/lib/models/item_class.dart
class ItemClass {
  final int id;
  final String name;
  final double basePrice;
  final double unitCost;
  final int washMinutes;

  ItemClass({
    required this.id,
    required this.name,
    required this.basePrice,
    this.unitCost = 0,
    this.washMinutes = 30,
  });

  factory ItemClass.fromJson(Map<String, dynamic> json) {
    return ItemClass(
      id: json['id'],
      name: json['name'],
      basePrice: (json['base_price'] as num).toDouble(),
      unitCost: (json['unit_cost'] as num?)?.toDouble() ?? 0,
      washMinutes: (json['wash_minutes'] as num?)?.toInt() ?? 30,
    );
  }
}